### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by gordysc**
- [x] Add a yarn.lock file
- [x] Fix existing linting errors
- [x] Upgrade ESLint to v5.3.X
- [x] Upgrade babel-eslint to 8.2.X
- [x] Upgrade eslint-plugin-react to 7.10.X
- [x] Fix NPM script (doesn't lint repository files appropriately)
- [x] Require Node v6.4+ (due to ESLint upgrade dropping support for older Node versions)

### CODE REVIEW

**Comment by gordysc** [comment]
Please let me know if I missed anything, if there's a rule we want to tweak, etc.

File: package.json [hunk 1]
```diff
@@ -1,21 +1,21 @@
 {
   "name": "@dialexa/eslint-config-dialexa",
-  "version": "4.1.1",
+  "version": "5.3.0",
```
Comment (ehis) [inline package.json]: Why `5.3.0` instead of `5.0.0`?

Comment (ehis) [reply package.json]: This is our npm version

Comment (gordysc) [reply package.json]: To match the ESlint version. I have no strong feelings about these if you want to go to 5.0.0

**Pull request merged**